### PR TITLE
Azure Monitor Exporter - Set types for Remote Dependency

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
Modified `AzureMonitorConverter.cs` to set RemoteDependency mappings
Addressing #17026

### CODE REVIEW

File: sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/AzMonListExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                }
+            }
+
+            return null;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static object[] GetTagValues(this AzMonList tagObjects, params string[] tagNames)
+        {
+            int? length = tagNames?.Count();
```
Comment (reyang) [inline sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/AzMonListExtensions.cs:29]: Curious - is it possible to have `tagNames == null`?

File: sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/AzMonListExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            if (length == null || length == 0)
+            {
+                return null;
+            }
+
+            object[] values = new object[(int)length];
+
+            for (int i = 0; i < tagObjects.Length; i++)
```
Comment (reyang) [inline sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/AzMonListExtensions.cs:37]: Curious, will `i < tagObject.Length` result in the `tagObjects.Length` being evaluated each time during the loop?

Comment (cijothomas) [reply sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/AzMonListExtensions.cs:37]: I think so.(havent verified though). we should extract tagObjects.Length to a variable above.

File: sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/AzMonListExtensions.cs (hunk 1 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/AzMonListExtensions.cs:29]: It's our internal implementation. `tagNames` will always have value, could remove this check.

File: sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/HttpHelper.cs [hunk 3]
```diff
@@ -20,136 +15,81 @@ internal static class HttpHelper
...
 
-                if (httpHost != null)
+                if (!string.IsNullOrWhiteSpace(httpHost))
                 {
                     var httpPortAndTarget = tagObjects.GetTagValues(SemanticConventions.AttributeHttpHostPort, SemanticConventions.AttributeHttpTarget);
 
                     if (httpPortAndTarget[0] != null && httpPortAndTarget[0].ToString() != "80" && httpPortAndTarget[0].ToString() != "443")
                     {
-                        url = $"{httpScheme}://{httpHost}:{httpPortAndTarget[0]}{httpPortAndTarget[1]}";
+                        var IsColon = (string.IsNullOrWhiteSpace(httpPortAndTarget[0]?.ToString()) ? null : Colon);
```
Comment (reyang) [inline sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/HttpHelper.cs:46]: The null check and `ToString()` part would probably require some clean up (e.g. in the if-condition we already made the validation that `httpPortAndTarget[0] != null`.

**Review by reyang — APPROVED** [review]
LGTM with some minor suggestions.

File: sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/AzMonListExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.OpenTelemetry.Exporter.AzureMonitor
+{
+    internal static class AzMonListExtensions
```
Comment (TimothyMothra) [inline sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/AzMonListExtensions.cs:10]: Why do these need to be extension methods?
I would expect AzMonList to have methods GetValue and GetValues as part of that class.

Comment (rajkumar-rangaraj) [reply sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/AzMonListExtensions.cs:10]: Very good call out, moved this as a part of `AzMonList`

Comment (TimothyMothra) [reply sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/AzMonListExtensions.cs:10]: Cool. please include some unit tests specifically for the NULL and Empty conditions. 
I saw lots of attention to those conditions in your implementation.

Comment (rajkumar-rangaraj) [reply sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/AzMonListExtensions.cs:10]: I moved this method from a different file to AzMonListExtensions.cs. This PR does not change any functionality to it. I will get the test covered for this functionality in subsequent PRs.

**Pull request merged**